### PR TITLE
[LLM] Unify Transformers and Native API

### PR DESCRIPTION
**Pull request by sgwhat**
## Description
This PR is primarily aimed at unifying the API Usage of BigDL LLM Transformers.

### 1. Why the change?
Unify the usage of the Transformers API.

### 2. User API changes
See https://github.com/analytics-zoo/nano/issues/508 for more details.

### 3. Summary of the change 
- Deprecate `BigdlNativeForCausalLM`.
- Add specific models like `LlamaForCausalLM` which supports native and transformers int4 meanwhile.

### 4. How to test?
- [x] Unit test
- [x] Document test

### CODE REVIEW

File: python/llm/src/bigdl/llm/transformers/modelling_bigdl.py [hunk 1]
```diff
@@ -20,58 +20,209 @@
 # only search the first bigdl package and end up finding only one sub-package.
 
 from bigdl.llm.utils.common import invalidInputError
+from .model import AutoModelForCausalLM
 
 
-class BigdlNativeForCausalLM:
```
Comment (jason-dai) [inline python/llm/src/bigdl/llm/transformers/modelling_bigdl.py:25]: don't remove old API, deprecate it.

**Comment by jason-dai** [comment]
Need to update examples

File: python/llm/src/bigdl/llm/transformers/modelling_bigdl.py (hunk 1 above)
Comment (sgwhat) [reply python/llm/src/bigdl/llm/transformers/modelling_bigdl.py:25]: Sure.

File: python/llm/src/bigdl/llm/transformers/modelling_bigdl.py [hunk 2]
```diff
@@ -20,58 +20,209 @@
...
-    """
+class LlamaForCausalLM:
 
     @classmethod
     def from_pretrained(cls,
                         pretrained_model_name_or_path: str,
-                        model_family: str = 'llama',
-                        dtype: str = 'int4',
+                        native: bool = True,
+                        dtype: str = "int4",
```
Comment (hkvision) [inline python/llm/src/bigdl/llm/transformers/modelling_bigdl.py:32]: This is only valid when native=True? If so, do we need to add something in the doc?

File: python/llm/src/bigdl/llm/transformers/modelling_bigdl.py [hunk 3]
```diff
@@ -20,58 +20,209 @@
...
+               , `gptneox` and `starcoder`.
+        :param cache_dir: (optional) This parameter will only be used when ``native`` is True
+               and ``pretrained_model_name_or_path`` is a huggingface checkpoint or hub repo id.
                It indicates the saving path for the converted low precision model.
-        :param tmp_path: (optional) Which path to store the intermediate fp16 model during the
-               conversion process. Default to `None` so that intermediate model will not be saved.
-        :param kwargs: keyword arguments which will be passed to the model instance
+        :param tmp_path: (optional) This parameter will only be used when ``native`` is True.
+               Which path to store the intermediate fp16 model during the conversion process.
+               Default to `None` so that intermediate model will not be saved.
```
Comment (hkvision) [inline python/llm/src/bigdl/llm/transformers/modelling_bigdl.py:50]: These two are not explicit in args?

**Comment by hkvision** [comment]
Need to check the api doc before merge: https://bigdl.readthedocs.io/en/latest/doc/LLM/index.html

File: python/llm/src/bigdl/llm/transformers/modelling_bigdl.py (hunk 2 above)
Comment (sgwhat) [reply python/llm/src/bigdl/llm/transformers/modelling_bigdl.py:90]: `native = True` -> Native int4; 
`native = False` -> Transformers int4

File: python/llm/src/bigdl/llm/transformers/modelling_bigdl.py (hunk 3 above)
Comment (sgwhat) [reply python/llm/src/bigdl/llm/transformers/modelling_bigdl.py]: Right, they are not specified, and maybe we can remove them?

**Comment by sgwhat** [comment]
Pending runner env fixed to merge PR, API docs and readme will be updated in a separated PR.

**Comment by jason-dai** [comment]
Shall we merge it now or after the demo? @shane-huang

File: python/llm/src/bigdl/llm/transformers/modelling_bigdl.py [hunk 4]
```diff
@@ -52,6 +53,9 @@ def from_pretrained(cls,
 
         :return: a model instance
         """
+        invalidInputError(False,
+                          "This API has been deprecated, "
+                          "please switch to the new API for sepcific models.")
```
Comment (hkvision) [inline python/llm/src/bigdl/llm/transformers/modelling_bigdl.py:58]: I think to make it deprecated, we shouldn't throw error, but raise a warning https://github.com/intel-analytics/BigDL/blob/25f569ff2e79c86976f3d4bc81f97f445ae0edb4/python/orca/src/bigdl/orca/tfpark/tfnet.py#L268 or mark as deprecated https://github.com/intel-analytics/BigDL/blob/25f569ff2e79c86976f3d4bc81f97f445ae0edb4/python/chronos/src/bigdl/chronos/autots/deprecated/config/recipe.py#L152 ?

File: python/llm/test/inference/test_call_models.py [hunk 5]
```diff
@@ -43,6 +45,11 @@ def test_llama_completion_with_stream_success(self):
         llm = Llama(self.llama_model_path, n_threads=self.n_threads)
         output = llm("What is the capital of France?", max_tokens=32, stream=True)
 
+    def test_llama_unify_native_api(self):
```
Comment (hkvision) [inline python/llm/test/inference/test_call_models.py:48]: rename to test_llama_for_causallm?

File: python/llm/test/inference/test_transformers_api.py [hunk 6]
```diff
@@ -90,5 +90,24 @@ def test_transformers_auto_model_for_speech_seq2seq_int4(self):
         res = 'Mr. Quilter is the apostle of the middle classes and we are glad to welcome his gospel.' in transcription[0]
         self.assertTrue(res)
 
+    def test_transformers_unify_api(self):
```
Comment (hkvision) [inline python/llm/test/inference/test_transformers_api.py:93]: test_transformers_chatglm_for_causal_lm?

**Review by hkvision — APPROVED** [review]

**Pull request merged**